backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/predicate/impl/ElasticsearchSimpleQueryStringPredicateBuilder.java

```diff
@@ -63,6 +66,38 @@ public void defaultOperator(BooleanOperator operator) {
 		}
 	}
 
+	@Override
+	public void enable(SimpleQueryFlag operation) {
+		if ( this.operations == null ) {
+			this.operations = EnumSet.noneOf( SimpleQueryFlag.class );
```

Comment:
If the function "disable" is used for the first time, it will set all to enabled and the indicated one to disable.

If the function "enable" is used for the first time, it will set all to disabled and the indicated one to enable.

Reply:
If you want my opinion, you could simplify the code (and tests!) a lot by offering only these two:

* `flags(SimpleQueryFlag ... flags)`
* `flags(Set<SimpleQueryFlag>)`

... and assuming that calling the `flags` method twice will just take into account the last call.

This is more adapted to the "fluent" nature of the DSL anyway, where you can't easily have if/else constructs between calls. Users you would just build your set of flags beforehand, then pass it to the DSL.

*Disabling*  some features, i.e. enabling "everything except this and that", is probably not a good idea in general: if you care about what you want enabled, you should probably care enough to explicitly tell us which ones should be enabled. But, it would still be possible:

```
.flags( EnumSet.complementOf( EnumSet.of( SimpleQueryFlag.PREFIX ) ) )
```

Reply:
Of course. So leave only these two functions proposed? If it doesn't use any functions then it's considered all enabled?

Reply:
Exactly. I believe this will make everything easier for you, and should be relatively succinct and easy to use in most cases.

Reply:
Well. I optimized these functions. It actually looks better.

Reply:
I built the "LuceneSimpleQueryParserPredicateIT" tests. But I have a problem with Jenkiens, local tests have good results, in Jenkiens it always strangely returns empty lists. I supported the JSON library there to load data for testing faster, but that's not the reason. I don't know what could be the reason Hudson is doing my tests?

Reply:
```
[Hits of LuceneSearchQueryImpl[query=+(+(string:hiber*)^0.5 +(string:robot)^0.5) #__HSEARCH_type:main, sort=null]] 
Expecting:
  <[]>
to contain exactly (and in same order):
  <[DocRef:IndexName_1_1/1_1_4, DocRef:IndexName_1_1/1_1_3]>
but could not find the following elements:
  <[DocRef:IndexName_1_1/1_1_4, DocRef:IndexName_1_1/1_1_3]>

	at org.hibernate.search.integrationtest.backend.lucene.search.LuceneSimpleQueryParserPredicateIT.searchParentSimpleQueryString(LuceneSimpleQueryParserPredicateIT.java:119)
```

Reply:
Yeah, it's because you're loading the data from the filesystem instead of loading it through the classloader. In your IDE, JARs are unzipped and their content loaded directly from the filesystem, so there's no difference, but when running tests from the commandline, JARs are zipped so the files cannot be simply accessed from the filesystem.

Don't worry about this: I'll take care of it before merging. As I mentioned in another PR, we usually avoid, in all the hundreds of tests we already have, to create a big dataset like that, and combine smaller datasets to generate the index data programatically instead. So, to preserve consistency, I'll probably convert your dataset to something smaller that does not rely on a JSON file in the end.

Reply:
Fine. But then you should write three tests, especially for the multi-value problem.

It turns out that the tests are helpful during programming. You can quickly identify the problem because it is possible to debug the test. Writing in json greatly speeds up test preparation. Of course, you can choose smaller models. I used such models because I wanted to check if the problem is solved in the master, flanttended and nested documents. After that, the json data samples read very well.

So if you can write a test quickly, there is no problem. Especially that such a test proves to be very helpful.